Windows/ChangeBPMWindow.xaml.cs

```diff
@@ -67,6 +67,7 @@ public partial class ChangeBPMWindow : Window {
         }
 
         private void dataBPMChange_CurrentCellChanged(object sender, EventArgs e) {
+            caller.mapEditor.currentMapDifficulty?.MarkDirty(); // This might be too aggressive - triggers even when just clicking on the data cell
```

Comment:
There should be some property in the `EventArgs` that lets you check if the new value is different from the old value. That should prevent the map from being marked dirty with just a click.

Reply:
I'll take a look later today.

Reply:
Reviewing the documentation, this event fires when the cell selection in the grid changes, so it's definitely not the right place to mark as dirty. For edits we already have RowEditEnding, for new rows we have AddingNewItem. One thing that's not covered if I remove it is the BPM change deletion, but I've handled that separately.